src/kernels/levelx/xminmax.opencl

```diff
@@ -0,0 +1,179 @@
+
+// =================================================================================================
+// This file is part of the CLBlast project. Author(s):
+//   Ekansh Jain
+//
+// This file contains the Xminmax kernel. It implements index of (absolute) min/max computation using
+// reduction kernels. Reduction is split in two parts. In the first (main) kernel the X vector is
+// loaded, followed by a per-thread and a per-workgroup reduction. The second (epilogue) kernel
+// is executed with a single workgroup only, computing the final result.
+//
+// =================================================================================================
+
+// Enables loading of this file using the C++ pre-processor's #include (C++11 standard raw string
+// literal). Comment-out this line for syntax-highlighting when developing.
+R"(
+
+// Parameters set by the tuner or by the database. Here they are given a basic default value in case
+// this kernel file is used outside of the CLBlast library.
+#ifndef WGS1
+  #define WGS1 64     // The local work-group size of the main kernel
+#endif
+#ifndef WGS2
+  #define WGS2 64     // The local work-group size of the epilogue kernel
+#endif
+
+// =================================================================================================
+
+// The main reduction kernel, performing the loading and the majority of the operation
+#if RELAX_WORKGROUP_SIZE == 1
+  __kernel
+#else
+  __kernel __attribute__((reqd_work_group_size(WGS1, 1, 1)))
+#endif
+void Xminmax(const int n,
+           const __global real* restrict xgm, const int x_offset, const int x_inc,
+           __global real* restrict mgm, __global unsigned int* igm) {
+  __local singlereal maxlm[WGS1];
+  __local singlereal minlm[WGS1];
+  __local unsigned int imaxlm[WGS1];
+  __local unsigned int iminlm[WGS1];
```

Comment:
An issue here is that this uses the tuning parameters from the `Xdot` routine, which don't use as much local memory as this routine. And thus this might lead to invalid kernels, depending on the values for WGS1. So I guess you'll need to introduce separate tuning variables as well. Sounds like quite some work, perhaps there is an easier way around this? Simply halving WGS1 here to be safe with local memory size but slightly less optimal for speed? I think that could work.